src/lf_apply.rs

```diff
@@ -126,70 +92,93 @@ unsafe fn backup_lpf<BD: BitDepth>(
             );
             row += stripe_h; // unmodified stripe_h for the 1st stripe
             stripe_h = 64 >> ss_ver;
-            src = src.offset(stripe_h as isize * BD::pxstride(src_stride as usize) as isize);
-            dst = dst.offset(n_lines as isize * BD::pxstride(dst_stride as usize) as isize);
+            src_offset =
+                (src_offset as isize + stripe_h as isize * BD::pxstride(src_stride)) as usize;
+            dst_offset =
+                (dst_offset as isize + n_lines as isize * BD::pxstride(dst_stride)) as usize;
+
             if n_lines == 3 {
-                BD::pixel_copy(
-                    slice::from_raw_parts_mut(dst, dst_w as usize),
-                    slice::from_raw_parts(
-                        &mut *dst.offset(-(BD::pxstride(dst_stride as usize) as isize))
-                            as *mut BD::Pixel,
-                        dst_w as usize,
-                    ),
-                    dst_w as usize,
-                );
-                dst = dst.offset(BD::pxstride(dst_stride as usize) as isize);
+                if dst_stride < 0 {
+                    let dst_tmp = dst.split_at_mut(dst_offset + BD::pxstride(-dst_stride) as usize);
+                    dst_tmp.0[dst_offset..][..dst_w as usize]
+                        .copy_from_slice(&dst_tmp.1[..dst_w as usize]);
+                } else {
+                    let dst_tmp = dst.split_at_mut(dst_offset);
+                    dst_tmp.1[..dst_w as usize].copy_from_slice(
+                        &dst_tmp.0[dst_offset - BD::pxstride(dst_stride) as usize..]
+                            [..dst_w as usize],
+                    );
+                }
```

Comment:
Could you rewrite this in a similar way you did the last one?  Like using `BD::pixel_copy`, and unifying some of the branches if possible (it was very clean how you did that in the last one).  And in general, for methods like `.split_at_mut`, I think it's clearer to destructure the tuple and give the halves names rather than using `.0` and `.1`?